jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java

```diff
@@ -216,4 +274,79 @@ protected String getSearchColumnName(ExamplesTable searchData) {
+        if (serverLogXmlReporterExtension != null) {
+            serverLogXmlReporterExtension.registerLogContent(key, logContent);
+        } else {
+            log.warn("ServerLogXmlReportExtension is not registered, no logs will be present in report.");
+        }
+    }
+
+    private void storeTemplatesLogs(Map<String, List<SshTemplate>> sshTemplates) {
+        sshTemplates.entrySet().stream().forEach(entry ->
+            entry.getValue().stream().forEach(sshTemplate -> {
+                MultiKey multiKey = new MultiKey(entry.getKey(),
+                    scenarioStart.toString(),
+                    scenarioEnd.toString(),
+                    sshTemplate.getSshSetting().getHostname() + ":" + sshTemplate.getSshSetting().getPort(),
+                    sshTemplate.getSshSetting().getLogPath());
+                try {
+                    String sshLog = sshTemplate.copyLog(scenarioStart, scenarioEnd).getLogContents();
+                    storeInExtension(multiKey, sshLog);
+                } catch (Exception e) {
+                    storeInExtension(multiKey, ExceptionUtils.getStackTrace(e));
+                }
+            })
+        );
+    }
+
+    private <T> Map<String, List<T>> getSshTemplatesForType(Class<T> clazz) {
+        Map<String, List<T>> sshTemplates = new HashMap<>();
+        String[] beanNames = beanFactory.getBeanNamesForType(clazz);
+        for (String beanName : beanNames) {
+            BeanDefinition bd = beanFactory.getMergedBeanDefinition(beanName);
+            if (bd instanceof RootBeanDefinition) {
+                Qualifier qualifier = ((RootBeanDefinition) bd).getResolvedFactoryMethod().getAnnotation(Qualifier.class);
+                if (sshTemplates.get(qualifier.value()) == null) {
+                    sshTemplates.put(qualifier.value(), new ArrayList<>());
+                }
+                sshTemplates.get(qualifier.value()).add((T) beanFactory.getBean(beanName));
+            }
+        }
+        return sshTemplates;
+    }
+
+    private Map<String, List<SshTemplate>> getSshTemplates() {
+        Map<String, List<SshTemplate[]>> sshTemplatesArray = getSshTemplatesForType(SshTemplate[].class);
+        Map<String, List<SshTemplate>> sshTemplates = getSshTemplatesForType(SshTemplate.class);
+
+        //merge both maps together to simple Map<String, List<SshTemplate>>
+        sshTemplatesArray.entrySet()
+            .stream()
+            .forEach(entry -> {
+                List<SshTemplate> flattenedSshTemplatesArray = entry.getValue()
+                    .stream()
+                    .flatMap(Arrays::stream)
+                    .collect(Collectors.toList());
+                if (sshTemplates.get(entry.getKey()) != null) {
+                    sshTemplates.get(entry.getKey()).addAll(flattenedSshTemplatesArray);
+                } else {
+                    sshTemplates.put(entry.getKey(), flattenedSshTemplatesArray);
+                }
+            });
```

Comment:
```
sshTemplatesArray.entrySet().stream()
            .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().stream().flatMap(Arrays::stream).collect(Collectors.toList())))
            .forEach((k, v) -> sshTemplates.merge(k, v, (v1, v2) -> {
                v2.addAll(v1);
                return v2;
            }));
```

Reply:
Thank you

Reply:
v1, v2 >>  list

Reply:
Variables renamed + changed to use biconsumer foreach.